lib/remote_configuration/lib/src/implementation/firebase_remote_configuration.dart

```diff
@@ -29,15 +30,33 @@ class FirebaseRemoteConfiguration extends RemoteConfiguration {
     return _remoteConfig.getBool(key);
   }
 
+  /// Initializes the remote configuration with the given default values and
+  /// fetches the remote configuration in the background.
+  ///
+  /// The default values are used if the remote configuration could not be
+  /// fetched.
+  ///
+  /// This method follows the "Load new values for next startup" strategy:
+  /// https://firebase.google.com/docs/remote-config/loading.
   @override
-  Future<void> initialize(Map<String, dynamic> defaultValues) async {
+  Future<void> initializeAndFetchInBackground(
+    Map<String, dynamic> defaultValues,
+  ) async {
     try {
       _remoteConfig = FirebaseRemoteConfig.instance;
       _remoteConfig.setDefaults(defaultValues);
-      _remoteConfig.setConfigSettings(RemoteConfigSettings(
+      _remoteConfig.setConfigSettings(
+        RemoteConfigSettings(
           fetchTimeout: const Duration(minutes: 1),
-          minimumFetchInterval: const Duration(hours: 3)));
-      await _remoteConfig.fetchAndActivate();
+          minimumFetchInterval: const Duration(hours: 3),
+        ),
+      );
+      // Activate the fetched remote config from the last fetch.
+      await _remoteConfig.activate();
+
+      // Fetch remote config in the background. They will be available on the
+      // next app start.
+      unawaited(_remoteConfig.fetch());
     } catch (e) {
       log("Error fetch remote config: $e");
     }
```

Comment:
Couldn't we just leave the initialize method with the old logic and add the `initializeAndFetchInBackground` method that just calls `unawaited(initialize)`?

Reply:
No strong need but I also don't see why we would have to remove initialize entirely... (If we leave it should we call it `intializeAndFetch` as well then?)

Reply:
I refactored the code to a new style. I removed all the activate and fetch logic from the `initialize` method (which makes for more sense that the `initialize` method only initializes), added a `fetch` and `activate` method to the package so that we can use the "Load new values for next startup" strategy the `plugin_initializations` file.